src/Field.tsx

```diff
@@ -41,13 +40,13 @@ export interface FieldProps<V = any> {
   form: FormikProps<V>; // if ppl want to restrict this for a given form, let them.
 }
 
-export interface FieldConfig {
+export interface FieldConfig<ExtraProps> {
   /**
    * Field component to render. Can either be a string like 'select' or a component.
    */
   component?:
     | string
-    | React.ComponentType<FieldProps<any>>
+    | React.ComponentType<ExtraProps & FieldProps<any>>
```

Comment:
This generic is what allows us have the additional props available on the `<Field />` inferred from the component props. Same idea should be applicable to the `<Formik />` component.